api/lib/opentelemetry/trace/tracer.rb

```diff
@@ -54,10 +72,12 @@ def start_root_span(name, attributes: nil, links: nil, start_timestamp: nil, kin
       #   +with_parent_context+.
       # @param [optional SpanContext] with_parent_context Explicitly managed. Overridden by
       #   +with_parent+.
+      ## @param [optional Context] with_context Explicitly managed context. Overridden by
+      #   +with_parent_context+ or +with_parent+
       #
       # @return [Span]
-      def start_span(name, with_parent: nil, with_parent_context: nil, attributes: nil, links: nil, start_timestamp: nil, kind: nil, sampling_hint: nil)
-        span_context = with_parent&.context || with_parent_context || current_span.context
+      def start_span(name, with_parent: nil, with_parent_context: nil, with_context: nil, attributes: nil, links: nil, start_timestamp: nil, kind: nil, sampling_hint: nil)
```

Comment:
The new parameter name is **really** confusing. We now have 3 parameters to specify the parent span context, and they're confusingly similarly named while accepting completely different argument types:
- `with_parent`: `Span`
- `with_parent_context`: `SpanContext`
- `with_context`: `Context`

The latter expands to 5 different options:
- implicit context current span's context
- implicit context extracted span context
- explicit context current span's context
- explicit context extracted span context
- invalid context

Reply:
We could probably get rid of `with_parent_context`. I don't see a use case where you'd have a span context without a `Context` or a `Span` any longer.

Reply:
I got rid of `with_context` and repurposed `with_parent_context` to take a `Context` instead of a `SpanContext`.